lib/galaxy/workflow/refactor/schema.py

```diff
@@ -190,6 +224,47 @@ class UpdateReportAction(BaseAction):
     ACTION_CLASSES_BY_TYPE[action_type] = action_class
 
 
-class RefactorRequest(BaseModel):
+class RefactorActions(BaseModel):
     actions: List[Action]
     dry_run: bool = False
+
+
+class RefactorActionExecutionMessageTypeEnum(str, Enum):
+    tool_version_change = 'tool_version_change'
+    tool_state_adjustment = 'tool_state_adjustment'
+    connection_drop_forced = 'connection_drop_forced'
+    workflow_output_drop_forced = 'workflow_output_drop_forced'
+
+
+class RefactorActionExecutionMessage(BaseModel):
+    message: str
+    message_type: RefactorActionExecutionMessageTypeEnum
+    # messages don't have to be bound to a step, but if they are should
+    # specify step_label and order_index below - these are the label and
+    # order_index before applying the refactoring. If connections are
+    # dropped this step reference should refer to the step with the
+    # previously connected input.
+    step_label: Optional[str]
+    order_index: Optional[int]
+
+    # messages don't have to be bound to a step with inputs, but if they are
+    # the input name should be specified here. Should be a prefixed name
+    # in the case of tool inputs (e.g. 'cond|repeat_0|input')
+    input_name: Optional[str]
+
+    # messages don't have to be bound to a step with inputs, but if they are
```

Comment:
I really like those comments. Can those become descriptions ?

Reply:
Done? It is much harder to read inline now but they will be pretty in an OpenAPI schema someday.

Reply:
Thanks!